Best practices/Serialization & Deserialization.md

```diff
@@ -0,0 +1,124 @@
+
+## System.Text.Json
+
+Serialization and deserialization are fundamental processes in software development, allowing for the conversion of objects to a format that can be easily stored or transmitted and then reconstructed later. In the .NET ecosystem, while there have been several libraries available for these operations, `System.Text.Json` has emerged as a modern,  high-performance option built directly into .NET Core.
+
+For performance-critical scenarios  using `System.Text.Json` source generators  can provide significant improvements by generating optimized serialization code at compile-time.
+
+### Basic Usage
+
+To convert an object into its JSON representation:
+
+```string jsonString = JsonSerializer.Serialize(yourObject);```
+
+To convert a JSON string back into an object:
+
+```YourObjectType obj = JsonSerializer.Deserialize<YourObjectType>(jsonString);```
+
+### JsonSerializerOptions
+
+`JsonSerializerOptions` is a class that provides a way to specify various serialization and deserialization behaviors. Here are some of the commonly used options:
+
+**PropertyNameCaseInsensitive**: When set to `true`, property name matching during deserialization is case-insensitive. This can be useful when dealing with JSON from sources that don't maintain consistent casing.
+
+**WriteIndented**: Determines whether the output JSON should be pretty-printed with indentation, which can be useful for debugging or human readability.
+
+**DefaultIgnoreCondition**: Determines the condition under which a property will be ignored during serialization. For instance, you can set it to `JsonIgnoreCondition.WhenWritingNull` to ignore properties with `null` values.
+
+**AllowTrailingCommas**: When set to `true`, the deserializer will tolerate trailing commas in the JSON.
+
+**MaxDepth**: Sets the maximum depth allowed when reading or writing JSON, protecting against deeply nested structures. This is primarily a safeguard against potential stack overflow exceptions, especially in recursive parsing scenarios. For example, setting **MaxDepth = 1** would give us an extremely shallow structure, meaning we are only handling top-level properties. This could be useful if we are sure that our JSON data should only have simple key-value pairs without any nested objects or arrays.
+
+**ReferenceHandler**: Controls how object references are managed during serialization/deserialization.
+There are primarily two built-in settings for the `ReferenceHandler` property:
+1. **ReferenceHandler.Default**: This is the default behavior, where object references are not preserved during serialization. If you have circular references in your object graph, serializing it with this setting will result in a stack overflow exception.
```

Comment:
Add an empty line between the text and the beginning of numbered block, the parser won't show the renumbered list as intended without it.